File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> counter = new HashMap<>();
        for (int i = 0; i < clothes.length; i++) {
            if (counter.containsKey(clothes[i][1])) {
                counter.put(clothes[i][1], counter.get(clothes[i][1]) + 1);
            } else {
                counter.put(clothes[i][1], 1);
            }
        }
        
        for (String key : counter.keySet()) {
            answer *= counter.get(key) + 1; // 해당 종류를 안입을 경우를 더해주고
        }
        
        answer -= 1; //아무것도 안입는 경우를 빼준다.
        return answer;
    }
}

File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    types = {}
    for i in range(len(clothes)):
        if clothes[i][1] not in types:
            types[clothes[i][1]] = 1
        else :
            types[clothes[i][1]] += 1
            
    for key in types:
        answer *= types[key] + 1
    
    answer -= 1

    return answer